pkg/proxy/iptables/proxier.go

```diff
@@ -643,7 +643,8 @@ func (proxier *Proxier) syncProxyRules() {
 
 	start := time.Now()
 	defer func() {
-		metrics.SyncProxyRulesLatency.Observe(metrics.SinceInMicroseconds(start))
+		metrics.SyncProxyRulesLatency.Observe(metrics.SinceInSeconds(start))
```

Comment:
Is this dropping precision? why?  I expect that most of the time this takes < 1 second, butthe difference between 0.2 seconds and 0.9 seconds is kind of meaningful.  It seems to be float so maybe it's just weird naming?

Reply:
@thockin In fact, `SinceInSeconds` (nanoseconds level) have higher precision than `SinceInMicroseconds` (microseconds level).
For example:
The result of this case:
```
fmt.Printf("SinceInMicroseconds: %v\nSinceInSeconds: %v\n", SinceInMicroseconds(time.Now().Add(-time.Second)), SinceInSeconds(time.Now().Add(-time.Second)))
```
is
```
SinceInMicroseconds: 1.000001e+06
SinceInSeconds: 1.000000806
```
The name `SinceInSeconds` is used to show its unit as `seconds`.
But we can discuss about this naming. cc @brancz

Reply:
We're not loosing precision as it's still a float64, this is just a consistency change, to make sure all metrics use the base unit, which is the best practice so it is easier for different metrics to be used with each other (this is part of the Prometheus instrumentation best practices, which among others includes using [seconds as time base unit](https://prometheus.io/docs/practices/naming/#base-units). Complying to the Prometheus best practices is one of the motivations for the [metrics overhaul KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/0031-kubernetes-metrics-overhaul.md#motivation), which this is part of.